structuralcodes/codes/mc2020/_concrete_limit_state_of_cracking.py

```diff
@@ -0,0 +1,1367 @@
+        )
+
+    theta_rad = math.radians(theta)
+    return (sigma_y + tau_xy / math.tan(theta_rad)) / rho_s_ef_y
+
+
+def beta_TS_x(
+    beta_TS: float, sr_max_theta: float, theta: float, sr_max_x: float
+) -> float:
+    """Calculate the empirical coefficient beta_TS for the x direction.
+
+    fib Model Code 2020, eqs. (30.5-20) and (30.5-21)
+
+    Args:
+        beta_TS (float): Base beta_TS coefficient.
+        sr_max_theta (float): Maximum crack spacing in the theta direction in
+            mm.
+        theta (float): Angle between the direction of reinforcement and the
+            direction of principal tensile stress in degrees.
+        sr_max_x (float): Characteristic crack spacing in the given direction x
+            in mm.
+
+    Returns:
+        float: Empirical coefficient beta_TS for the specified direction.
+
+    Raises:
+        ValueError: If any input is invalid.
+    """
+    if sr_max_theta <= 0 or sr_max_x <= 0 or not (0 <= theta <= 90):
+        raise ValueError(
+            'Invalid input values for empirical coefficient calculation.'
+        )
+
+    theta_rad = math.radians(theta)
+
+    return beta_TS * (sr_max_theta / math.sin(theta_rad)) / sr_max_x
+
+
+def beta_TS_y(
+    beta_TS: float, sr_max_theta: float, theta: float, sr_max_y: float
+) -> float:
+    """Calculate the empirical coefficient beta_TS for the y direction.
+
+    fib Model Code 2020, eqs. (30.5-20) and (30.5-21)
+
+    Args:
+        beta_TS (float): Base beta_TS coefficient.
+        sr_max_theta (float): Maximum crack spacing in the theta direction in
+            mm.
+        theta (float): Angle between the direction of reinforcement and the
+            direction of principal tensile stress in degrees.
+        sr_max_y (float): Characteristic crack spacing in the given direction y
+            in mm.
+
+    Returns:
+        float: Empirical coefficient beta_TS for the specified direction.
+
+    Raises:
+        ValueError: If any input is invalid.
+    """
+    if sr_max_theta < 0 or sr_max_y <= 0 or not (0 <= theta <= 90):
+        raise ValueError(
+            'Invalid input values for empirical coefficient calculation.'
+        )
+
+    theta_rad = math.radians(theta)
+
+    return beta_TS * (sr_max_theta / math.cos(theta_rad)) / sr_max_y
+
+
+def eps_sm_eps_cm_theta(
+    eps_sm_x_eps_cm_x: float, eps_sm_y_eps_cm_y: float, eps_2: float
+) -> float:
+    """Computes the principal relative mean strain given the relative strain in
+    the x-y directions.
+
+    fib Model Code 2020, eqs. (30.5-18)
+
+    Args:
+        eps_sm_x_eps_cm_x (float): Relative strain in the x direction.
+        eps_sm_y_eps_cm_y (float): Relative strain in the y direction.
+
+    Returns:
+        float: The principal mean strain.
+
+    Raises:
+        ValueError: If any input is invalid
+    """
+    if (
+        eps_sm_x_eps_cm_x < 0
+        or eps_sm_y_eps_cm_y < 0
+        or eps_sm_y_eps_cm_y < eps_2
```

Comment:
I am not seein this limitation of `eps_sm_y_eps_cm_y` beeing greater or equal than `eps_2`